DiscordBot/Commands/ICommand.cs

```diff
@@ -1,11 +1,16 @@
-﻿using System.Threading.Tasks;
-using DSharpPlus.Entities;
-
-namespace DiscordBot.Commands
-{
-    // Commands that only send a message in response
-    interface ICommand
-    {
-        Task Run(DiscordMessage msg);
-    }
-}
+﻿using System.Threading.Tasks;
+using DiscordBot.Backends;
+
+namespace DiscordBot.Commands
+{
+    // Commands that only send a message in response
+    public interface ICommand
+    {
+        Task Run(IContext e);
+
+        Task OnMessage(IContext e)

```

Comment:
If this function is not called for every message, maybe a better name would be `OnRegularMessage`?

Reply:
yeah